### PR TITLE
OpenFoam-paraview-catalyst

### PR DESCRIPTION
**Pull request by sbna**
Merged PR #8752 and #8753 in a unique PR as we agreed. PR #8752 and #8753 can be closed.
Updated openfoam, paraview and catalyst packages and added the new package of-catalyst
to couple openfoam with catalyst to perform in situ visualization.

### CODE REVIEW

**Comment by dorier** [comment]
I see `-DVTK_USE_SYSTEM_EXPAT:BOOL=ON`, doesn't this need Expat to be declared as a dependency?

**Comment by sbna** [comment]
> I see `-DVTK_USE_SYSTEM_EXPAT:BOOL=ON`, doesn't this need Expat to be declared as a dependency?

They asked me to always use it as an external dependency.

File: var/spack/repos/builtin/packages/of-catalyst/package.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    This offering is part of the community repository supported by OpenCFD Ltd,
+    producer and distributor of the OpenFOAM software via www.openfoam.com,
+    and owner of the OPENFOAM trademark.
+    OpenCFD Ltd has been developing and releasing OpenFOAM since its debut
+    in 2004.
+    """
+
+    # Currently only via git
+    homepage = "https://develop.openfoam.com/Community/catalyst"
+    gitrepo  = "https://develop.openfoam.com/Community/catalyst.git"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/of-catalyst/package.py]: If you name this `git` you won't need to specify `git` in the versions below.

File: var/spack/repos/builtin/packages/of-catalyst/package.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    variant('full', default=False, description='Build against paraview (full) or catalyst (light)')
+
+    depends_on('openfoam-com@1806', when='@1806', type=('build', 'link', 'run'))
+    depends_on('openfoam-com@develop', when='@develop', type=('build', 'link', 'run'))
+    depends_on('catalyst@5.5:', when='~full')
+    depends_on('paraview@5.5:+osmesa~qt', when='+full')
+
+    def setup_environment(self, spack_env, run_env):
+        run_env.prepend_path('LD_LIBRARY_PATH', join_path(self.prefix,
+                             'lib'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/of-catalyst/package.py]: This should be done automatically.

File: var/spack/repos/builtin/packages/of-catalyst/package.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    @property
+    def root_cmakelists_dir(self):
+        """The relative path to the directory containing CMakeLists.txt
+
+        This path is relative to the root of the extracted tarball,
+        not to the ``build_directory``. Defaults to the current directory.
+
+        :return: directory containing CMakeLists.txt
+        """
+        return join_path(self.stage.source_path, join_path('src', 'catalyst'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/of-catalyst/package.py]: You can replace this whole thing with:
```python
root_cmakelists_dir = 'src/catalyst'
```

File: var/spack/repos/builtin/packages/of-catalyst/package.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    def cmake_args(self):
+        """Populate cmake arguments for ParaView."""
+        cmake_args = [
+            '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY:PATH=%s' % join_path(
+                self.stage.source_path,
+                'spack-build'),
+            '-DCMAKE_INSTALL_PREFIX:PATH=%s' % self.prefix
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/of-catalyst/package.py]: This should be set by the `CMakePackage` base class.

**Pull request closed without merging**